pkg/controller/job/backoff_utils_test.go

```diff
@@ -355,6 +355,27 @@ func TestGetFinishedTime(t *testing.T) {
 			},
 			wantFinishTime: defaultTestTime,
 		},
+		"Pod with init container and all containers terminated": {
```

Comment:
Maybe name the test to reflect the intention for testing sidecar containers, say 
"Pod with init container representing a sidecar container, all containers terminated"

Reply:
Should we?
In a real cluster, I think only sidecar containers can be terminated after main containers terminated.
However, in code perspective, we don't check if a init container is a sidecar container here. We simply check all init container statuses, assuming normal init containers will terminate before main containers and do no harm to `finishTime`.
So as we don't care if a initContainer status belongs to a sidecar container, I think maybe to need to reflect the intention for testing sidecar containers?

Reply:
No strong view here. I think the code when an init container is stopped after the main container may look suspicious to some folks working on Job. Still, maybe reflecting this at the title of the PR is enough, so we can find out by looking at the lines history.

Reply:
> I think the code when an init container is stopped after the main container may look suspicious to some folks working on Job. 

Yes, I agree...

I added comments to both UT and `getFinishTimeFromContainers`, I think this should be enough.

Reply:
sgtm